tests/test_helpers/test_base.py

```diff
@@ -34,9 +34,13 @@ def test_locate(self):
         """Test the locate function to locate modules."""
         cwd = os.getcwd()
         os.chdir(os.path.join(CUR_PATH, ".."))
-        assert locate("packages.fetchai.connections.gym") is not None
-        assert locate("packages.fetchai.connections.non_existing_connection") is None
+        gym_package = locate("packages.fetchai.connections.gym")
+        non_existing_package = locate(
+            "packages.fetchai.connections.non_existing_connection"
+        )
         os.chdir(cwd)
+        assert gym_package is not None
+        assert non_existing_package is None
```

Comment:
note, the previous implementation was dangerous as it would have left the working dir changed on assertion failure